syntax = "proto2";
package revolve.msgs;
import "parameter.proto";

message NeuralConnection {
  required string src = 1;
  required string dst = 2;
  required double weight = 3;
}

message Neuron {
  required string id = 1;
  required string layer = 2;
  required string type = 3;
  optional string partId = 4;
  repeated Parameter param = 5;
}

message NeuralNetwork {
  repeated Neuron neuron = 1;
  repeated NeuralConnection connection = 2;
}

message ModifyNeuralNetwork {
  repeated string remove_hidden = 1;
  repeated Neuron add_hidden = 2;
  repeated Neuron set_parameters = 4;
  repeated NeuralConnection set_weights = 3;
}
